disks-and-volumes.md

```diff
@@ -0,0 +1,315 @@
+# Volumes and Disks
+
+## Disks
+
+Like all other vm devices a `spec.domain.devices.disks` element has a mandatory
+`name`. Furhter it has a mandatory `volumeName` entry which references a volume
+inside `spec.volumes`.
+
+A disk can be made accessible via four different types:
+
+ * disk
+ * cdrom
+ * floppy
+ * lun
+
+### lun
+
+A `lun` disk will expose the volume as a LUN device to the vm.
+
+A minimal example which attaches a `PersistentVolumeClame` named `mypvc` as a
+`lun` device to the vm:
+
+```
+metadata:
+  name: testvm-ephemeral
+apiVersion: kubevirt.io/v1alpha1
+kind: VirtualMachine
+spec:
+  domain:
+    resources:
+      requests:
+        memory: 64M
+    devices:
+      disks:
+      - name: mypvcdisk
+        volumeName: mypvc
+        # This makes it a lun device
+        lun: {}
```

Comment:
I like the empty structs for - empty structs :)

(lun: {} vs lun:)

I'll see that I have an eye on this in examples in general.

Reply:
`lun:` is equivalent to `lun: null`, whereas `lun: {}` is an empty struct/dict. So having `{}` means it is present, whereas `lun:` is the same as not specifying it.